lib/generators/deimos/schema_class/templates/schema_record.rb.tt

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# autogenerated by Deimos, do not modify
+module Deimos
+  # :nodoc:
+  class <%= schema_classname(@current_schema) %> < SchemaRecord
+<%- fields.each do |field| -%>
+    # @return [<%= deimos_field_type(field) %>]
+    attr_accessor :<%= field.name %>
+<% end -%>
+<%- unless @current_schema.name =~ /\-key$/ -%>
+    # @return [Object] An optional payload key
+    attr_accessor :payload_key
+<% end -%>
+
+    # @override
+    def initialize(**kwargs)
+      super()
+      args = kwargs.with_indifferent_access
+<%- fields.each do |field| -%>
+      <%= "@#{field.name} = args[:#{field.name}]" %>
+<% end -%>
+      @payload_key = args[:payload_key]
+    end
+
+    # @override
+    def self.initialize_from_payload(payload)
+      return unless payload.present?
+
+      args = {}
+      payload.each do |key, value|
+<%- unless @special_field_initialization.present? -%>
+        args[key.to_sym] = value
+<%- else -%>
+        args[key.to_sym] = case key.to_sym
+<%- @special_field_initialization.each do |initialize_string, fields| -%>
+                           when <%= fields.join(', ') %>
+                             <%= initialize_string %>
+<% end -%>
+                           else
+                             value
+                           end
+<%- end -%>
+      end
+      self.new(**args)
+    end
```

Comment:
I feel like this might be simpler if you just generate a `def #{field}` and `def #{field}=` method rather than relying on `attr_accessor`. This works but it seems a bit odd.

It would also make it easier to set default values! 😁

Reply:
hahaha good idea

Reply:
Using `attr_readers` and created custom `def #{field}=` methods for fields that need this special handling!

using `attr_accessors` for everything else!